src/kem/xyber768d00.rs

```diff
@@ -0,0 +1,281 @@
+        // enforce_equal_len above.
+        let x = <<X25519HkdfSha256 as KemTrait>::PrivateKey as Deserializable>::from_bytes(
+            &encoded[..32],
+        )?;
+        let k = encoded[32..].try_into().unwrap();
+
+        Ok(PrivateKey { x, k })
+    }
+}
+
+/// An X25519-Kyber768 public key. This holds an X25519 public key and a Kyber768 public key.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[cfg_attr(feature = "serde_impls", derive(serde::Serialize, serde::Deserialize))]
+#[doc(hidden)]
+pub struct PublicKey {
+    x: <X25519HkdfSha256 as KemTrait>::PublicKey,
+    #[cfg_attr(feature = "serde_impls", serde(with = "serde_big_array::BigArray"))]
+    k: pqc_kyber::PublicKey,
+}
+
+/// An X25519-Kyber768 private key. This holds an X25519 private key and a Kyber768 private key.
+#[derive(Clone)]
+#[cfg_attr(feature = "serde_impls", derive(serde::Serialize, serde::Deserialize))]
+#[doc(hidden)]
+pub struct PrivateKey {
+    x: <X25519HkdfSha256 as KemTrait>::PrivateKey,
+    #[cfg_attr(feature = "serde_impls", serde(with = "serde_big_array::BigArray"))]
+    k: pqc_kyber::SecretKey,
+}
+
+/// Holds the content of an encapsulated secret. This is what the receiver uses to derive the
+/// shared secret. Since this is a hybrid KEM, it holds a DH encapped key and a Kyber encapped key.
+#[derive(Clone)]
+#[cfg_attr(feature = "serde_impls", derive(serde::Serialize, serde::Deserialize))]
+#[doc(hidden)]
+pub struct EncappedKey {
+    x: <X25519HkdfSha256 as KemTrait>::EncappedKey,
+    #[cfg_attr(feature = "serde_impls", serde(with = "serde_big_array::BigArray"))]
+    k: [u8; 1088],
+}
+
+impl ConstantTimeEq for PrivateKey {
+    fn ct_eq(&self, other: &Self) -> Choice {
+        self.x.to_bytes().ct_eq(&other.x.to_bytes()) & self.k.ct_eq(&other.k)
+    }
+}
+
+impl PartialEq for PrivateKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.ct_eq(other).into()
+    }
+}
+
+impl Eq for PrivateKey {}
+
+#[doc = "Represents X25519Kyber768Draft00 v2"]
+pub struct X25519Kyber768Draft00;
+
+impl KemTrait for X25519Kyber768Draft00 {
+    // X25519Kyber768Draft00 v2 §5: Nsecret of X25519Kyber768Draft00 is 64
+    #[doc(hidden)]
+    type NSecret = typenum::U64;
+
+    type EncappedKey = EncappedKey;
+    type PublicKey = PublicKey;
+    type PrivateKey = PrivateKey;
+
+    const KEM_ID: u16 = 0x30;
+
+    // X25519Kyber768Draft00 v2 §3.3
+    // def DeriveKeyPair(ikm):
+    //   dkp_prk = LabeledExtract("", "dkp_prk", ikm)
+    //   seed = LabeledExpand(dkp_prk, "sk", 32 + 64)
+    //   seed1 = seed[0:32]
+    //   seed2 = seed[32:96]
+    //   sk1, pk1 = X25519KEM.DeriveKeyPair(seed1)
+    //   sk2, pk2 = Kyber768Draft00.DeriveKeyPair(seed2)
+    //   return (concat(sk1, sk2), concat(pk1, pk2))
+
+    /// Deterministically derives a keypair from the given input keying material and ciphersuite
+    /// ID. The keying material should have as many bits of entropy as it extracts for its seeds,
+    /// i.e., 768 bits.
```

Comment:
This "768 bits" doesn't make sense to me. See [this comment](https://github.com/rozbb/rust-hpke/pull/43/files#r1265260721).

Reply:
Fixed